core/interfaces/i_focusable_node.ts

```diff
@@ -59,6 +59,21 @@ export interface IFocusableNode {
    * This has the same implementation restrictions as onNodeFocus().
    */
   onNodeBlur(): void;
+
+  /**
+   * Indicates whether this node allows focus. If this returns false then none
+   * of the other IFocusableNode methods will be called.
+   *
+   * Note that special care must be taken if this dynamically changes its return
+   * value over the lifetime of the node as certain environment conditions could
+   * affect the focusability of this node's DOM element (such as whether the
+   * element has a positive or zero tabindex). Also, changing from a true to a
+   * false value while the node holds focus would not change the current focus
+   * of the node, and may result in some of the node's functions being called
+   * later on when defocused (since it was previously considered focusable at
+   * the time of being focused).
+   */
+  canBeFocused(): boolean;
```

Comment:
Update docs to include a returns, and reconsider the caveats in `INavigable.isNavigable` to see if some of those should be moved here.

Reply:
Copied over the main blob from `INavigable` and adjusted it to better tailor toward focus, and provide accessibility-specific advice. I did drop the 'Field'-specific example since it would be contradicted by my additional consideration that even disabled or read-only fields may still benefit from being focusable. I also mentioned invisible elements since that's actually the only case today where we actually disable focusability (for the keyboard navigation plugin a la https://github.com/google/blockly-keyboard-experimentation/pull/502.

Reply:
I think I would probably remove "disabled, read-only" from the list of examples for why you might want this to return false, as those cases should still be focusable for screenreader users most likely. I think it really is just the purely visual decoration / no visual representation case.

Reply:
I agree, though I was trying to maintain some parity with the existing documentation.

@gonfunko and @rachel-fenichel do you have any thoughts as this documentation would be giving explicit guidance to implementations?